ui/app/build-configs/detail/details-tab/pncBuildConfigDetailsEditor.js

```diff
@@ -60,14 +64,31 @@
       // Ensure this components copy of the BC can't be updated from outside.
       $ctrl.buildConfig = angular.copy($ctrl.buildConfig);
       $ctrl.originalBuildType = $ctrl.formData.general.buildType;
+      $ctrl.formData.productVersion = $ctrl.productVersion ? $ctrl.productVersion : null;
+      $ctrl.formData.product = $ctrl.productVersion ? $ctrl.productVersion.product : null;
     };
 
+   /*Check version data and remove it if product is changed or not selected*/
+      $ctrl.checkVersionData = () => {
+        if (!$ctrl.formData.product || ($ctrl.formData.productVersion && $ctrl.formData.product.id !== $ctrl.formData.productVersion.product.id)) {
+          $ctrl.formData.productVersion = null;
+        }
+      };
+
     function submit() {
       $ctrl.working = true;
       var buildConfig = toBuildConfig($ctrl.formData, $ctrl.buildConfig);
 
       BuildConfigResource.safePatchRemovingParameters($ctrl.buildConfig, buildConfig).$promise
-          .then(resp => $ctrl.onSuccess({ buildConfig: resp}))
+          .then(resp => {
+            $ctrl.onSuccess({ buildConfig: resp});
+            $state.go('projects.detail.build-configs.detail', {
```

Comment:
Why are you navigating to the same state? Especially with reload true this causes everything to be reloaded from the backend.

Reply:
Hi Alex,
The purpose of this action is to solve the issue that when the user edited the product version information, it can't be updated until the page was refreshed, see https://github.com/project-ncl/pnc/pull/3188#discussion_r459250436
I will try to find some solution to update the data without refreshing the page or using the "=" symbol to pass the product version information either.

Thanks

Reply:
Fixed and committed to this PR